libeos-payg/manager.c

```diff
@@ -1329,6 +1337,14 @@ file_load_cb (GObject      *source_object,
       self->key_bytes = g_bytes_new_take (g_steal_pointer (&data), data_len);
       data_len = 0;
     }
+  else if (g_file_equal (file, self->account_id_file))
+    {
+      if (data != NULL)
+        {
+          g_free ((gchar *) self->account_id);
```

Comment:
The field shouldn't be declared `const`. Remove that qualifier and the cast won't be needed.

Reply:
The same cast in `finalize` can be removed too.

Reply:
Ok, I was trying to keep the changes to a minimum, but I agree that makes sense.

Reply:
hum, actually, I think this is correct. `const <type> *` will mean the contents of the array are immutable, which is what we want (although I don't think it matters too much in this case either way). Except if this is any different since the type here is a gchar instead of regular char, but I bet not.

I'll keep it as-is for now.

Reply:
But you're not treating the pointed to value immutably here since you're freeing the data and the location this is pointing to in the heap is something else entirely. It's not the end of the world, but it seems pretty clearly wrong. It's trying to express that the value can't be changed, but `const` just doesn't capture those semantics in this use case.

Reply:
Ok, I think my confusion here is that I never realized that freeing a pointer is considered modifying the memory area pointed by it. And the assignment is just changing the address being pointed to, not the contents of the memory area.

Anyway, I've fixed it now.